src/main/scala/net/akmorrow13/endive/pipelines/SolverPipeline.scala

```diff
@@ -125,10 +125,19 @@ object SolverPipeline extends Serializable with Logging {
 
     if (conf.negativeSamplingFreq < 1.0) {
       println("NEGATIVE SAMPLING")
-      val positives = trainFeaturizedWindows.filter(_.labeledWindow.label > 0)
-      val negatives = trainFeaturizedWindows.filter(_.labeledWindow.label == 0).sample(false, conf.negativeSamplingFreq)
+      val positives = trainFeaturizedWindows.filter(_.labeledWindow.label > 0).cache()
+      val negatives = trainFeaturizedWindows.filter(_.labeledWindow.label == 0).sample(false, conf.negativeSamplingFreq).cache()
       trainFeaturizedWindows = positives.union(negatives)
     }
+      val numTrainPositives = trainFeaturizedWindows.filter(_.labeledWindow.label > 0).count()
+      val numTrainNegatives = trainFeaturizedWindows.filter(_.labeledWindow.label == 0).count()
+
+      val numTestPositives = valFeaturizedWindowsOpt.map(_.filter(_.labeledWindow.label > 0).count()).getOrElse(0.0)
+      val numTestNegatives = valFeaturizedWindowsOpt.map(_.filter(_.labeledWindow.label == 0).count()).getOrElse(0.0)
+
+      println(s"NUMBER OF TRAIN (POS, NEG) is ${numTrainPositives}, ${numTrainNegatives}")
+      println(s"NUMBER OF TEST (POS, NEG) is ${numTestPositives}, ${numTestNegatives}")
+
```

Comment:
If we are not using these values can we remove the lines here? It takes about 20 min for these lines